### PR TITLE
Bugfix: `cdm_select_tbl()` with multi-FKs between 2 tables

### PR DESCRIPTION
**Pull request by TSchiefer**
currently on master:
``` r
library(dm)
#> 
#> Attaching package: 'dm'
#> The following object is masked from 'package:stats':
#> 
#>     filter
cdm_nycflights13(cycle = TRUE) %>% cdm_draw()
```

``` r
cdm_select_tbl(cdm_nycflights13(cycle = TRUE), flights, airports) %>% cdm_draw()
```

<sup>Created on 2019-11-04 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

in this branch:
``` r
library(dm)
#> 
#> Attaching package: 'dm'
#> The following object is masked from 'package:stats':
#> 
#>     filter
cdm_nycflights13(cycle = TRUE) %>% cdm_draw()
```

``` r
cdm_select_tbl(cdm_nycflights13(cycle = TRUE), flights, airports) %>% cdm_draw()
```

<sup>Created on 2019-11-04 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### CODE REVIEW

File: R/select-tbl.R [hunk 1]
```diff
@@ -72,7 +72,7 @@ filter_recode_table_fks <- function(def, selected) {
 }
 
 filter_recode_table <- function(data, selected) {
-  idx <- match(selected, data$table, nomatch = 0L)
+  idx <- map(unname(selected), ~which(. == data$table)) %>% flatten_int()
```
Comment (krlmlr) [inline R/select-tbl.R]: The `match()` was a daft idea.

```suggestion
  idx <- data$table %in% selected
```

**Comment by TSchiefer** [comment]
Argh, no, sorry, it doesn't work with `%in%` since we need the indices for ensuring the order...

**Comment by TSchiefer** [comment]
I can't think of a nicer solution than the one you already saw, that works in all cases. Please feel free to suggest something though.

File: R/select-tbl.R [hunk 2]
```diff
@@ -72,7 +72,7 @@ filter_recode_table_fks <- function(def, selected) {
 }
 
 filter_recode_table <- function(data, selected) {
-  idx <- match(selected, data$table, nomatch = 0L)
+  idx <- map(unname(selected), ~which(data$table == .)) %>% flatten_int()
```
Comment (krlmlr) [inline R/select-tbl.R]: - create an `ordered()` from `data$table` with `levels = selected`
- remove all `NA`
- sort

Comment (krlmlr) [reply R/select-tbl.R]: Alternatively, implement two functions, one for `def`, one for the fks?

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**